### ISSUE TITLE
Implementation of Mask RCNN for Fridge/Can Detection

### ISSUE THREAD

**Issue by Nike1421** [UCI-SAP-Capstone-2024/Proof-of-Performance-Validation#14]
In the previous meeting on 04/04/2024, it was decided during discussion that we should review the performance of Object Detection Models for our problem statement. A good marker for the same would be the detection of cans in a fridge / cans from an image.

The first model we aim to implement the same is Mask RCNN. We aim to use Transfer Learning to do so.

Tasks include:
1. Getting the dataset
2. Annotating the various objects in the images in the dataset
3. Training the model on the dataset
4. Evaluating the performance

**Comment by puneater** [comment]
Mask-RCNN posed a lot of compatibility issues.
1. TensorFlow does not support GPU for Windows
2. Mask-RCNN was developed on old versions of TensorFlow and Keras, which are no longer available (both on pip and conda)
3. After multiple unsuccessful attempts to run on new versions of TensorFlow and Keras we decided to discard Mask-RCNN, now implementing Detectron2
